### PR TITLE
26743 utilities in Nexus geometry

### LINKED ISSUE #26743: Changes to FileResource in Geometry Tests
Important changes required for `FileResource`
   1. Has no implementation file
   1. Cannot switch easily to a debug/preserve file mode.
   1. Naming clash with `ScopedFile` already in TestHelpers - We are confusing the community

### PR DESCRIPTION
**Pull request by TakudzwaMakoni**
**Description of work.**

**Do not merge before** #26751 and #26734

additional tests in Nexus geometry and utilities in file resource and NexusFileReader as required by the issue below
**To test:**

Fixes #26743 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by DavidFair** [comment]
@TakudzwaMakoni is this blocked by #26734 , or can it be merged separately?

**Comment by TakudzwaMakoni** [comment]
> @TakudzwaMakoni is this blocked by #26734 , or can it be merged separately?

that's right, it's waiting for changes to be merged with #26734, ask @OwenArnold  for further details

**Comment by OwenArnold** [comment]
@DavidFair There are some outstanding review changes and some merge conflicts to be dealt with. We would still appreciate you reviewing this PR when it is done.

**Comment by DavidFair** [comment]
@OwenArnold I'm still happy to review this once the changes and merge conflicts are resolved. 
Could you @ me when this is ready?

**Comment by TakudzwaMakoni** [comment]
@DavidFair are you able to review this today if and when you are free?

File: Framework/NexusGeometry/src/NexusGeometrySave.cpp
Comment (DavidFair) [inline Framework/NexusGeometry/src/NexusGeometrySave.cpp]: Why has this been unmarked as const? We only set it during initialisation currently?

File: Framework/NexusGeometry/src/NexusGeometrySave.cpp
Comment (DavidFair) [inline Framework/NexusGeometry/src/NexusGeometrySave.cpp]: Currently this class definition goes public, private public. Could we combine it all into a single private and public section (as it reads like two separate classes)

Additionally is there any reason we now support copying? 
Looking at the existing tests we currently don't use copies (might have missed something), and I would say that if there were copies created it's probably because someone didn't pass by ref

**Review by DavidFair — CHANGES_REQUESTED** [review]
A nice clean up moving the FileResourceHeader implementation into a .cpp file (every little bit to speed up compiling helps).

Unit tests ran successfully for me too.

There are some minor outstanding clean-up opportunities and questions outstanding, but overall I'm happy with this

File: Framework/NexusGeometry/src/NexusGeometrySave.cpp
Comment (TakudzwaMakoni) [reply Framework/NexusGeometry/src/NexusGeometrySave.cpp]: was caused by a previosu merge, fixed

File: Framework/NexusGeometry/src/NexusGeometrySave.cpp
Comment (TakudzwaMakoni) [reply Framework/NexusGeometry/src/NexusGeometrySave.cpp]: also caused by a previous merge, fixed.

**Review by DavidFair — APPROVED** [review]
I'm happy with this, as previously noted this tidies up some test related functions and moves an implementation out of a header into a separate source file.
It works locally for me still so lets :shipit:

**Comment by OwenArnold** [comment]
@mantidproject/gatekeepers could someone merge this soon please. It's holding up other work.

**Pull request merged**
